heat/core/linalg.py

```diff
@@ -68,7 +77,7 @@ def dot(a, b, out=None):
         raise NotImplementedError("ht.dot not implemented for N-D dot M-D arrays")
 
 
-def matmul(a, b):
+def matmul(a, b, no_resplit=False):
```

Comment:
default value for resplitting input tensors should be False i believe, meaning that no_resplit=True

Reply:
I think that it should be True. if we are targeting the usage on large datasets i think that we should try to get as much speed as possible. The downside being that it makes the learning curve steeper for users. @Markus-Goetz what do you think?

Reply:
@Markus-Goetz  Don't worry about it, @coquelin77 and I discussed this. We came to the conclusion that the best way would be to make the storage-wise more efficient version the default (aka no_resplit=False), offer the option to make the less efficient, but more expected and convenient method (not resplitting the inputs) optional via the flag and update the Docs to make it clear to the user.... so that they are warned
@rainman110 , @lehr-fa FYI